camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -222,8 +226,24 @@ public void run() throws Exception
       {
         System.err.println("unable to process " + topicName + " skipping...");
         e.printStackTrace();
-        executorService.shutdown();
-        throw e;
+      }
+    }
+
+    Future<?> completedTask;
+    int remainingTaskCount = tasksToComplete.size();
+
+    while (remainingTaskCount > 0) {
+      completedTask = completionService.take();
+      remainingTaskCount--;
+
+      try {
+          completedTask.get();
+      } catch (ExecutionException e) {
+          // If a task completed with an exception shutdown all other tasks and exit
+          // re-raise the exception to make the parent job fail.
+          executorService.shutdown();
```

Comment:
`shutdownNow` [does that as well](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html). However [a few lines down](https://github.com/vinted/camus/pull/26/files#diff-34802ddc212164b4c2a5af563768e90bR257) we'll be waiting for their termination or bailing out after timeout anyway.